### ISSUE TITLE
Loss functions interface in model training manager

### ISSUE THREAD

**Issue by PeterCalifano** [PeterCalifano/pyTorchAutoForge#30]
Current handling of losses in model training manager is way too experimental and error prone. A formalization of the interface as well as the expectations by the training manager and the internal auto-evaluation pipeline for monitoring and logging is really required.
A generalization is also needed to ensure training manager can work smoothly with tasks different from vector-to-vector or image-to-vector regression.
